pil-analyzer/src/pil_analyzer.rs

```diff
@@ -343,7 +358,7 @@ impl PILAnalyzer {
             expressions.push((expr, constr_function_statement_type()));
         }
 
-        let inferred_types = infer_types(definitions, &mut expressions)?;
+        let inferred_types = infer_types(definitions, &mut expressions, struct_declarations)?;
```

Comment:
You pass the definitions, why do you need to pass the structs as well?

Reply:
Because definitions no longer have what I need. I have to somehow keep the info inside`FunctionValueDefinition::StructDeclaration`, because they are the ones that contain the type of the Struct fields. 
At this point, definitions has only Expressions.